### DISCUSSION TITLE
Run Dockerized Jupyter with Dokku

### DISCUSSION THREAD

**Discussion by ohld** [dokku/dokku#5847] [Q&A]
# TL;DR

1. Blank screen after login page when deployed with `dokku git:from-image ...`. 
2. No issues when I run docker image without Dokku

--- 

I want to run a jupyter notebook server on my machine where Dokku is installed and my other apps are hosted. My goal is to run python notebook scripts not on my laptop but on my VPS.

I decided to try **official Jupyter images** from https://jupyter-docker-stacks.readthedocs.io/en/latest/. 

This is how I deployed them:
```
dokku apps:create jupyter 

# persistant storage for files
dokku storage:mount jupyter "${PWD}":/home/jovyan/work

dokku git:from-image jupyter jupyter/scipy-notebook:latest

# jupyter runs on :8888 by default:
dokku proxy:ports-set jupyter http:80:8888

# add https
dokku letsencrypt:enable jupyter
```

Everything deployed successfully on https://jupyter.mydomain.com. I opened the login page (rendered perfectly), entered access token which was copied from `dokku logs -t jupyter`. 

But then suddenly I see blank screen:
<img width="1512" alt="image" src="https://github.com/dokku/dokku/assets/5613295/78978ace-7098-45b1-9f8d-64bdae29909c">

When I run this docker image **without Dokku** I don't have any issues: everything works as intended. 

I thought that the problem was with websocket ports but **this didn't solve** the problem:
```
dokku proxy:ports-add jupyter ws:80:8888 wss:443:8888
```

So I'm asking for community help. I really want to stay in Dokku ecosystem: it is great ❤️

**Comment by josegonzalez** [comment]
We have a tutorial [here](https://dokku.com/tutorials/other/using-websockets-in-dokku/) on how to setup an app that uses websockets for a given process.

**Comment by ohld** [comment]
Thanks. Now it is clear how to setup ws by committing nginx config to a repo. _(but I had a microservice with ws endpoint that worked smoothly without any additional nginx config)_. 

But is there a way to do that with pre-built docker image?

**Reply by josegonzalez** [reply]
No, you'll need to have that custom nginx.conf.sigil committed to the WORKDIR in the built image.

**Reply by ohld** [reply]
Thanks for the fast reply.

I just doublechecked that there is no separate [process for websocket in Jupyter](https://github.com/jupyter/docker-stacks/blob/main/base-notebook/start-notebook.sh). Just `jupyter`. 

I'm not sure that there is an issue with ws connection. At least, I can't see any ws-related issue in Jupyter logs when I run using Dokku.

**Reply by josegonzalez** [reply]
What is in the browser console?

**Reply by ohld** [reply]
Just some 404s for static assets (screenshot above) which also happens with a regular Docker run (without Dokku). Nothing unusual, that's why I created this Discussion.
